### PR TITLE
feat: experimental method to download results by model tag

### PR DESCRIPTION
**Pull request by nankolena**
### Linked issue(s)

Resolves KOL-6900
Corresponding backend change: https://github.com/kolenaIO/shipyard/pull/2450

CI is expected to fail before the above change is in production. Planning to cut a minor version after 1.36 release.

### What change does this PR introduce and why?

- During `upload_results`, allow optionally specifying a list of string tags to be associated with the model. A model can be associated with multiple tags, and a tag can by associated with multiple models.
- Creating a method under experimental folder to support downloading results by specifying the model tag. Technically speaking, the backend change supports directly downloading results even if a tag is associated with multiple models. That however will make the function signature nested and complicated. Given that the immediate use case we see is a one-to-one mapping between a model and its human-readable tag, limiting the function by inspecting that only 1 model is associated with the provided tag and keeping the same return type as `download_results`. In case multiple models share the same tag, the exception message will contain the model names to help users identify which ones to use.

Also verified docs update:
image

### Please check if the PR fulfills these requirements

- [x] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [x] Relevant tests for the changes have been added
- [x] Relevant docs have been added / updated

### CODE REVIEW

File: .circleci/continue_config.yml
Comment (munkyshi) [inline .circleci/continue_config.yml:450]: Let's revert this as soon as #683 merges

Comment (nankolena) [reply .circleci/continue_config.yml:450]: Yup, created a ticket in linear to keep track: https://linear.app/kolena/issue/KOL-7444/use-small-resource-instance-for-client-repo-tests

File: kolena/_experimental/dataset/evaluation.py
Comment (munkyshi) [inline kolena/_experimental/dataset/evaluation.py]: I don't think typing like `tuple` and `list` existed until 3.9, which means we need to use `typing.Tuple` and `typing.List` as long as we support 3.8.

https://peps.python.org/pep-0585/

File: kolena/_experimental/dataset/evaluation.py
Comment (munkyshi) [inline kolena/_experimental/dataset/evaluation.py:52]: We should add a docstring to indicate that this is currently restricted to if the tag is unique to that model

File: .circleci/continue_config.yml
Comment (munkyshi) [reply .circleci/continue_config.yml:450]: Whoops, wrong line.

File: .circleci/continue_config.yml
Comment (munkyshi) [inline .circleci/continue_config.yml]: Let's revert this as soon as https://github.com/kolenaIO/kolena/pull/683 merges. Also worth adding a comment saying so, assuming we can do it without an extra CI trigger.

**Review by munkyshi — APPROVED** [review]
Some small nits, but approving to unblock after they are addressed.

**Pull request merged**
